osu.Framework/Timing/InterpolatingFramedClock.cs

```diff
@@ -100,11 +101,15 @@ public void ProcessFrame()
 
                 if (IsInterpolating)
                 {
-                    // apply time increase from interpolation.
+                    // Apply time increase from interpolation.
                     currentTime += realtimeClock.ElapsedFrameTime * Rate;
-                    // if we differ from the elapsed time of the source, let's adjust for the difference.
-                    // TODO: this is frame rate depending, and can result in unexpected results.
-                    currentTime += (framedSourceClock.CurrentTime - currentTime) / 8;
+
+                    // Then check the post-interpolated time.
+                    // If we differ from the current time of the source, gradually approach the ground truth.
+                    //
+                    // The remaining error halves every halfTime ms.
+                    // This may need further tweaking to be less discernible by users (upwards, likely?).
+                    currentTime = Interpolation.DampContinuously(currentTime, framedSourceClock.CurrentTime, 50, realtimeClock.ElapsedFrameTime);
```

Comment:
```suggestion
                    currentTime = Interpolation.DampContinuously(currentTime, framedSourceClock.CurrentTime, 50, Math.Abs(realtimeClock.ElapsedFrameTime));
```

Maybe just to be safe? Otherwise exponent could go negative, and some values (e.g. `-0.5`) are imaginary.

We're missing this in many places osu!-side, fwiw. But not everywhere, e.g.: https://github.com/ppy/osu/blob/a2bf8e398807a45fba99a14f03b063a48a5192f4/osu.Game.Rulesets.Osu/Mods/OsuModMagnetised.cs#L78-L79

Reply:
Right, I suppose that makes sense.